src/utils/companyNumberSanitizer.ts

```diff
@@ -0,0 +1,37 @@
+import { provide } from 'inversify-binding-decorators'
+
+@provide(CompanyNumberSanitizer)
+export default class CompanyNumberSanitizer {
+  private readonly COMPANY_NUMBER_SIZE: number = 8
+
+  public sanitizeCompany(companyNumber: string): string {
+    if (!companyNumber) {
+      throw new Error('Company number is required')
+    }
+
+    companyNumber = companyNumber.toUpperCase()
+    companyNumber = this.stripWhitespaces(companyNumber)
```

Comment:
consider creating dedicated variables for each step, rather than overwriting the original variable. Immutability is always preferred.